meteor/server/api/showStyles.ts

```diff
@@ -94,6 +95,26 @@ export async function insertShowStyleVariantInner(
 		_rundownVersionHash: '',
 	})
 }
+export async function insertShowStyleVariantWithBlueprint(
+	context: MethodContext | Credentials,
+	showStyleBaseId: ShowStyleBaseId,
+	blueprintConfig: IBlueprintConfig,
+	_id: ShowStyleVariantId,
+	_rundownVersionHash: string,
+	name?: string
+): Promise<ShowStyleVariantId> {
+	const access = await ShowStyleContentWriteAccess.anyContent(context, showStyleBaseId)
+	const showStyleBase = access.showStyleBase
+	if (!showStyleBase) throw new Meteor.Error(404, `showStyleBase "${showStyleBaseId}" not found`)
+
+	return ShowStyleVariants.insertAsync({
+		_id: _id,
+		showStyleBaseId: showStyleBaseId,
+		name: name || 'Copied variant',
```

Comment:
This method shouldn't know which flow it is being used in and therefore doesn't know that it's copying variants. What if I want to use this method in another flow that inserts showStyleVariants but has nothing to do with copying. Then this name will be incorrect.

If `name` is required then it shouldn't be an optional parameter. (Atleast not here)

Reply:
I actually made that change, but it got removed with the issue-split. Good catch, i'm changing it here as i do see the point.